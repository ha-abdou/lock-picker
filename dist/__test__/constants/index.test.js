"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const __1 = __importDefault(require(".."));
describe("Constants: ", () => {
    test("string", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", "xxx");
        })).toMatchSnapshot();
    });
    test("number", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", 111);
        })).toMatchSnapshot();
    });
    test("boolean", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", true);
        })).toMatchSnapshot();
    });
    test("null", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", null);
        })).toMatchSnapshot();
    });
    test("undefined", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", undefined);
        })).toMatchSnapshot();
    });
    test("array", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", ["sss", 777, false, null, undefined]);
        })).toMatchSnapshot();
    });
    test("json", () => {
        const exp = `constant`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addConst("constant", {
                number: 111,
                string: "111",
                boolean: false,
                array: ["sss", 777, false, null, undefined],
                object: { ee: 3 },
                null: null,
                undefined: undefined,
            });
        })).toMatchSnapshot();
    });
});
//# sourceMappingURL=index.test.js.map