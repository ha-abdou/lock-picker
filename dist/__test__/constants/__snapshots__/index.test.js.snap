// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Constants:  array 1`] = `
Array [
  Array [
    "sss",
    777,
    false,
    null,
    undefined,
  ],
]
`;

exports[`Constants:  boolean 1`] = `
Array [
  true,
]
`;

exports[`Constants:  json 1`] = `
Array [
  Object {
    "array": Array [
      "sss",
      777,
      false,
      null,
      undefined,
    ],
    "boolean": false,
    "null": null,
    "number": 111,
    "object": Object {
      "ee": 3,
    },
    "string": "111",
    "undefined": undefined,
  },
]
`;

exports[`Constants:  null 1`] = `
Array [
  null,
]
`;

exports[`Constants:  number 1`] = `
Array [
  111,
]
`;

exports[`Constants:  string 1`] = `
Array [
  "xxx",
]
`;

exports[`Constants:  undefined 1`] = `
Array [
  undefined,
]
`;
